server/services/placeOnReportService.ts

```diff
@@ -225,4 +230,17 @@ export default class PlaceOnReportService {
       statementComplete: draftAdjudication.incidentStatement?.completed || false,
     }
   }
+
+  async getAssociatedStaffDetails(staffMembers: StaffSearchByName[], user: User): Promise<StaffSearchByNameEnhanced[]> {
+    const token = await this.hmppsAuthClient.getSystemClientToken(user.username)
+
+    const getLocationName = async (staffMember: StaffSearchByName) => {
+      if (staffMember.activeCaseLoadId === 'CADM_I') return { ...staffMember, currentLocation: 'Central Admin' }
+
+      const locationName = await new PrisonApiClient(token).getAgency(staffMember.activeCaseLoadId)
+      return { ...staffMember, currentLocation: locationName?.description }
+    }
+
+    return Promise.all(staffMembers.map((staffMember: StaffSearchByName) => getLocationName(staffMember)))
```

Comment:
It's not great to make network calls for every search row, especially if we aren't paginating.

What we tend to do to minimise network calls is to remove duplicates before making the network calls.